scripts/record-cached-http.ts

```diff
@@ -0,0 +1,356 @@
+
+    if (!data || !data.interactions || !Array.isArray(data.interactions)) {
+      return yamlContent;
+    }
+
+    // Sensitive header keys to sanitize
+    const sensitiveKeys = [
+      "x-api-key",
+      "authorization",
+      "x-auth-token",
+      "api-key",
+      "x-anthropic-api-key",
+      "x-openai-api-key",
+    ];
+
+    // Process each interaction
+    data.interactions.forEach((interaction: any) => {
+      if (interaction.request?.headers) {
+        sensitiveKeys.forEach((key) => {
+          const lowerKey = key.toLowerCase();
+          Object.keys(interaction.request.headers).forEach((headerKey) => {
+            if (headerKey.toLowerCase() === lowerKey) {
+              interaction.request.headers[headerKey] = ["cached-http-with-invalid-key"];
+            }
+          });
+        });
+      }
+    });
+
+    return yaml.dump(data, { lineWidth: -1, noRefs: true });
+  } catch (error) {
+    console.warn(
+      `Warning: Failed to parse YAML, returning original content: ${error instanceof Error ? error.message : String(error)}`
+    );
+    return yamlContent;
+  }
+}
+
+/**
+ * Set up the virtual environment using uv sync in cachedHTTP directory
+ */
+async function setupVenv(
+  cachedHTTPDir: string,
+  dryRun: boolean
+): Promise<{ success: boolean; error?: string }> {
+  if (dryRun) {
+    console.log(`[DRY RUN] Would sync venv in: ${cachedHTTPDir}`);
+    return { success: true };
+  }
+
+  // Ensure cachedHTTP directory exists
+  if (!existsSync(cachedHTTPDir)) {
+    return { success: false, error: `cachedHTTP directory not found: ${cachedHTTPDir}` };
+  }
+
+  // Use uv sync to create venv and install dependencies from pyproject.toml
+  console.log("Setting up virtual environment with uv sync...");
+  const syncResult = runCommand("uv", ["sync"], cachedHTTPDir);
+  if (!syncResult.success) {
+    return { success: false, error: syncResult.error || "Failed to sync venv" };
+  }
+
+  return { success: true };
+}
+
+/**
+ * Find Python files matching the pattern
+ */
+async function findPythonFiles(pattern: string): Promise<string[]> {
+  const files = await glob(pattern, {
+    cwd: process.cwd(),
+    absolute: true,
+  });
+  return files.filter((file) => file.endsWith(".py"));
+}
+
+/**
+ * Process a single Python file
+ */
+async function processFile(
+  pythonFile: string,
+  workDir: string,
+  cachedHTTPDir: string,
+  projectRoot: string,
+  dryRun: boolean
+): Promise<ProcessResult> {
+  try {
+    // Calculate relative path from project root
+    const relativePath = relative(projectRoot, pythonFile);
+    const recordingPath = resolve(workDir, relativePath);
+    const recordingDirPath = dirname(recordingPath);
+
+    // Read original file
+    const originalContent = await readFile(pythonFile, "utf-8");
+
+    // Calculate YAML path - Use absolute path so VCR.py creates cassettes in the right place
+    // regardless of current working directory
+    const yamlPath = recordingPath + ".yaml";
```

Comment:
I think it might be nice to have all the yaml in a sibling directory? Makes it easier to do things like ensure fresh regeneration of all the cassettes (by `rm -rf` the cassette dir), etc. Upstream in mirascope/mirascope I really like that the cassettes and snapshots are in separate directories from the tests. It is a little less important in this case admittedly since the src files are themselves non-canonical and copied from upstream... however in the medium term when we migrate this into the monorepo, the checked in source will be the copy in `./python/examples` and we won't want to pollute that file with snapshots. So building out the little bit of piping to put the cassettes in a distinct location is worth it imo.

Reply:
Sounds good.